CefSharp.Example/RequestEventHandler/RequestEventHandler.cs

```diff
@@ -0,0 +1,157 @@
+            OnCertificateErrorEvent?.Invoke(this, args);
+
+            //TODO: check if disposing the callback is intended like that
+            if (!args.Callback.IsDisposed)
+            {
+                args.Callback.Dispose();
+            }
+            return args.ContinueAsync;
+        }
+
+        public void OnPluginCrashed(IWebBrowser browserControl, IBrowser browser, string pluginPath)
+        {
+            var args = new OnPluginCrashedEventArgs(browserControl, browser, pluginPath);
+            OnPluginCrashedEvent?.Invoke(this, args);
+        }
+
+        public CefReturnValue OnBeforeResourceLoad(IWebBrowser browserControl, IBrowser browser, IFrame frame, IRequest request, IRequestCallback callback)
+        {
+            var args = new OnBeforeResourceLoadEventArgs(browserControl, browser, frame, request, callback);
+            OnBeforeResourceLoadEvent?.Invoke(this, args);
+
+            if (!args.Callback.IsDisposed)
+            {
+                args.Callback.Dispose();
+            }
+            return args.ResourceLoadHandling;
+        }
+
+        public bool GetAuthCredentials(IWebBrowser browserControl, IBrowser browser, IFrame frame, bool isProxy, string host, int port, string realm, string scheme, IAuthCallback callback)
+        {
+            var args = new GetAuthCredentialsEventArgs(browserControl, browser, frame, isProxy, host, port, realm, scheme, callback);
+            GetAuthCredentialsEvent?.Invoke(this, args);
+
+            if (!args.Callback.IsDisposed)
+            {
+                args.Callback.Dispose();
+            }
+
+            return args.ContinueAsync;
+        }
+
+        public void OnRenderProcessTerminated(IWebBrowser browserControl, IBrowser browser, CefTerminationStatus status)
+        {
+            var args = new OnRenderProcessTerminatedEventArgs(browserControl, browser, status);
+            OnRenderProcessTerminatedEvent?.Invoke(this, args);
+        }
+
+        public bool OnQuotaRequest(IWebBrowser browserControl, IBrowser browser, string originUrl, long newSize, IRequestCallback callback)
+        {
+            var args = new OnQuotaRequestEventArgs(browserControl, browser, originUrl, newSize, callback);
+            OnQuotaRequestEvent?.Invoke(this, args);
+
+            if (!args.Callback.IsDisposed)
+            {
+                args.Callback.Dispose();
+            }
+            return args.ContinueAsync;
+        }
+
+        public void OnResourceRedirect(IWebBrowser browserControl, IBrowser browser, IFrame frame, IRequest request, ref string newUrl)
+        {
+            var args = new OnResourceRedirectEventArgs(browserControl, browser, frame, request, new StringBuilder(newUrl));
+            OnResourceRedirectEvent?.Invoke(this, args);
+            //TODO: test if the string is really changed by using a StringBuilder (should work afaik ;) )
```

Comment:
`//TODO: test if the string is really changed by using a StringBuilder (should work afaik ;) )`